drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -205,18 +231,21 @@ static int spi_mcux_configure(const struct device *dev,
 
 		if (!device_is_ready(config->clock_dev)) {
 			LOG_ERR("clock control device not ready");
+			pm_policy_device_power_lock_put(dev);
 			return -ENODEV;
 		}
 
 		/* Get the clock frequency */
 		if (clock_control_get_rate(config->clock_dev,
 					   config->clock_subsys, &clock_freq)) {
+			pm_policy_device_power_lock_put(dev);
 			return -EINVAL;
 		}
 
 		if (spi_cfg->slave > SPI_CHIP_SELECT_COUNT) {
 			LOG_ERR("Slave %d is greater than %d",
 				    spi_cfg->slave, SPI_CHIP_SELECT_COUNT);
+			pm_policy_device_power_lock_put(dev);
```

Comment:
I suggest at this point since we have a lot of copy pasted error handling code, to make an `error` label at the bottom of function, set a `ret` value, and `goto error` so that we can have all the error handling code in one place. this is a common practice btw